polyply/src/minimizer.py

```diff
@@ -166,11 +172,19 @@ def target_function(positions):
         return energy
 
     opt_results = scipy.optimize.minimize(target_function, positions, method='L-BFGS-B',
-                                          options={'ftol':0.001, 'maxiter': 100})
-
+                                          options={'ftol': 0.001, 'maxiter': 100})
     positions = opt_results['x'].reshape((-1, 3))
     positions = renew_vs(positions, block, atom_to_idx)
     for node_key, idx in atom_to_idx.items():
         coords[node_key] = positions[idx]
 
-    return opt_results['success'], coords
+    max_fun = 0
+    for inter_type, tol in tolarance.items():
+        max_fun += tol**2. * len(block.interactions[inter_type])
```

Comment:
As a user I would interpret the tolerance differently. Instead of aggregating it all on one heap (where a very good bond could compensate a very bad dihedral or vice versa), I would expect the tolerance to be a limit *for every single interaction*. i.e. make sure that all bonds are within tolerance, and that all angles are, etc. Any interaction (not type!) not within tolerance should trigger failure.

Reply:
Yeah you're right probably not a very smart idea. I'll change this